src/definition/email/IPreEmailSent.ts

```diff
@@ -0,0 +1,25 @@
+import { IEmailDescriptor, IPreEmailSentContext } from '.';
+import { IHttp, IModify, IPersistence, IRead } from '../accessors';
+import { AppMethod } from '../metadata';
+
+/**
+ * Event interface that allows apps to
+ * register as a handler of of the `IPreEmailSent`
+ * event.
+ *
+ * This event is trigger before the mailer sends
+ * an email.
+ *
+ * To prevent the email from being sent, you can
+ * throw an error with a message specifying the
+ * reason for rejection.
+ */
+export interface IPreEmailSent {
+    [AppMethod.EXECUTE_PRE_EMAIL_SENT](
+        context: IPreEmailSentContext,
+        read: IRead,
+        http: IHttp,
+        persis: IPersistence,
+        modify: IModify,
+    ): Promise<IEmailDescriptor>;
```

Comment:
I am assuming this can modify the message? If so, maybe add the suffix `Modify` like we have for the messages and rooms. That way people know it's a method that can mutate the data

Reply:
We're moving away from that naming convention for events, to consolidate them into `Pre` or `Post` only. So this can modify _and_ prevent the event by throwing an error.

Reply:
Sounds like a good thing for v2 ;)